### PR TITLE
Document union_find_replacet

### PR DESCRIPTION
**Pull request by romainbrenguier**
In particular, the documentation for make_union was outdated.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 75b8386e42f25fdd50d8baa7b6a9f027794adf78).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89282695

File: src/util/union_find_replace.cpp [hunk 1]
```diff
@@ -8,13 +8,13 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
...
 
-/// Keeps a map of symbols to expressions, such as none of the mapped values
-/// exist as a key
-/// \param a: an expression of type char array
-/// \param b: an expression to map it to, which should be either a symbol
-///             a string_exprt, an array_exprt, an array_of_exprt or an
-///             if_exprt with branches of the previous kind
-/// \return the new mapped value
+/// Merge the set containing `a` and the set containing `b`.
+/// It is guaranteed that if `b` never appears as the first argument of a
```
Comment (smowton) [inline src/util/union_find_replace.cpp]: Does this function provide that guarantee, or require it from the caller?

Comment (kroening) [reply src/util/union_find_replace.cpp]: I think that guarantee is too obscure/non-standard to provide.
To make anything the root, use re_root()

Comment (romainbrenguier) [reply src/util/union_find_replace.cpp]: I removed this part of the comment, because changing the implementation may break this.

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: df01f72c45a6bf79b6cc22fedfd89f659601a013).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89637573
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Pull request merged**